llama-index-core/llama_index/core/program/llm_program.py

```diff
@@ -116,7 +116,7 @@ async def acall(
         llm_kwargs = llm_kwargs or {}
         if self._llm.metadata.is_chat_model:
             messages = self._prompt.format_messages(llm=self._llm, **kwargs)
-
+            messages = self._llm._extend_messages(messages)
```

Comment:
I don't really insist on using extend_messages, I just used it because I realized that it allready implements what I want😊
As far as I understand, the messages are just prepended with the allready stored system prompt, so it doesn't really make the LLM more stateful, does it? (or do You mean it changes behaviour slightly if it has a system prompt? this could probably be mitigated by setting a default system-prompt like 'You are a helpful assistant' or something like that.)

Personally, I really like the idea to be able to just give the system prompt when I create the LLM. Its just a convenient way for me to get an LLM that allready has a system prompt included and is ready to go.

Reply:
But then what if the LLM receives a prompt that already has a system prompt in it? Now we have to handle system prompts from both sides, which is confusing (and hence, why I'd rather remove it from the LLM object, but I feel like its stuck in a limbo state now)

Anyways, not really disagreeing with the PR here, but just pointing out some complexities